### PR TITLE
Add Vault setup ADR

### PR DESCRIPTION
**Pull request by iverberk**

### CODE REVIEW

File: docs/adr/0018-vault-setup.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# 18. vault-setup
+
+Date: 2021-10-06
+
+## Status
+
+Status: Proposal on 2021-10-06
+Builds on [0008-vault.md](0008-vault.md) on 2021-10-06
```
Comment (arminc) [inline docs/adr/0018-vault-setup.md:8]: I think it relates to 0007 as well

File: docs/adr/0018-vault-setup.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+## Decision
+
+### Authentication
+
+We will configure Vault with two authentication mechanisms: OIDC and JWT.
+
+The OIDC mechanism is used by human operators to access Vault. The OIDC implementation is handled by the Opsteady Azure AD. The OIDC token contains the claims that will grant access to the Vault with an appropriate policy. The primary claim which determines the level of access, is is the 'groups' claim. As an example, if you are a member of the 'platform-admin' group, you will be granted access to the Vault with full administrative privileges.
+
+The JWT mechanism is used by Github actions CI/CD. Vault will validate the presented JWT token with the Github actions issuer and allow access to the Vault with an appropriate policy for CI/CD purposes. This policy does not grant full administrative privileges and is primarly used to generate appropriate credentials and read component configuration.
```
Comment (arminc) [inline docs/adr/0018-vault-setup.md:23]: We need to record somewhere (story or security & compliance) that the CI/CD users gets more rights than needed now

File: docs/adr/0018-vault-setup.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+For AWS, Vault generates a temporary access key for a role in the target account. This role grants the necessary permissions to manage the infrastructure in the AWS account.
+
+### Network Access
+
+The information stored in Vault is highly sensitive and should be protected at all costs from unauthorized access. The above mentioned authentication mechanisms are the main barrier for access to the Vault but the endpoint for Vault, which is an HTTP API service, can also be shielded on the network level.
+
+We are going to expose the Vault endpoint on the Internet while we are building up the platform. Putting the endpoint on a private network complicates the CI/CD setup and overall management of the platform. We understand that this is a risk but we accept this risk for now. Once the Opsteady platform matures, and we move towards onboarding of customers, we will put Vault (and related security sensitive infrastructure like Azure Key Vault) on a private network with gated access controls (proxy) in place. The exact setup for this still needs to be determined.
```
Comment (arminc) [inline docs/adr/0018-vault-setup.md:41]: Should be added to security & compliance

File: docs/adr/0018-vault-setup.md (hunk 1 above)
Comment (iverberk) [reply docs/adr/0018-vault-setup.md:8]: Added it to the list.

File: docs/adr/0018-vault-setup.md (hunk 3 above)
Comment (iverberk) [reply docs/adr/0018-vault-setup.md:41]: This is added to the document.

**Pull request merged**